art/attacks/evasion/carlini.py

```diff
@@ -558,26 +561,31 @@ def __init__(
         self._targeted = targeted
         self.learning_rate = learning_rate
         self.max_iter = max_iter
-        self.max_halving = max_halving
-        self.max_doubling = max_doubling
-        self.eps = eps
-        self.batch_size = batch_size
+        self.decrease_factor = decrease_factor
+        self.initial_const = initial_const
+        self.largest_const = largest_const
+        self.const_factor = const_factor
         self.verbose = verbose
         self._check_params()
 
         # There is one internal hyperparameter:
         # Smooth arguments of arctanh by multiplying with this constant to avoid division by zero:
         self._tanh_smoother = 0.999999
 
-    def _loss(self, x_adv: np.ndarray, target: np.ndarray) -> Tuple[np.ndarray, np.ndarray]:
+    def _loss(
+        self, x_adv: np.ndarray, target: np.ndarray, x, const, tau
+    ) -> Tuple[np.ndarray, np.ndarray, np.ndarray, np.ndarray]:
         """
         Compute the objective function value.
 
-        :param x_adv: An array with the adversarial input.
+        :param x_adv: An array with the adversarial examples.
         :param target: An array with the target class (one-hot encoded).
-        :return: A tuple holding the current predictions and overall loss.
+        :param x: Benign samples.
+        :param  const: Current constant `c`.
+        :param tau: Current limit `tau`.
+        :return: A tuple of current predictions, total loss, logits loss and regularisation loss.
         """
-        z_predicted = self.estimator.predict(np.array(x_adv, dtype=ART_NUMPY_DTYPE), batch_size=self.batch_size)
+        z_predicted = self.estimator.predict(np.array(x_adv, dtype=ART_NUMPY_DTYPE), batch_size=1)
```

Comment:
Why is it necessary to hard-code a single prediction?

Reply:
I have set it to 1 because `x_adv` will always be a single sample and to express this explicitly.

Reply:
Thanks for the clarification. Given that it is not immediately clear why we choose `batch_size=1`, should we add a comment at an appropriate location in the code to mention why we explicitly use a batch size of one?